### PR TITLE
feat: Export Markdown to file system

### PR DESCRIPTION
**Pull request by rileyhawk1417**
# Description

This feature is meant to export the Markdown file to the Documents folder of the file system.
The PR is meant to address issue #411.

### CODE REVIEW

File: frontend/app_flowy/lib/workspace/application/doc/share_bloc.dart [hunk 1]
```diff
@@ -33,8 +36,24 @@ class DocShareBloc extends Bloc<DocShareEvent, DocShareState> {
   ExportData _convertDeltaToMarkdown(ExportData value) {
     final result = deltaToMarkdown(value.data);
     value.data = result;
+    writeFile(result);
     return value;
   }
+
+  Future<String> get _localPath async {
+    final dir = await getApplicationDocumentsDirectory();
```
Comment (appflowy) [inline frontend/app_flowy/lib/workspace/application/doc/share_bloc.dart:44]: Please use appFlowyDocumentDirectory instead. The path is different in different integration modes.

`Future<Directory> appFlowyDocumentDirectory() async {
  Directory documentsDir = await getApplicationDocumentsDirectory();

  switch (integrationEnv()) {
    case IntegrationMode.develop:
      return Directory('${documentsDir.path}/flowy_dev').create();
    case IntegrationMode.release:
      return Directory('${documentsDir.path}/flowy').create();
    case IntegrationMode.test:
      return Directory("${Directory.current.path}/.sandbox");
  }
}`

**Comment by appflowy** [comment]
@rileyhawk1417 Is the user notify the data was exported to local?

File: frontend/app_flowy/lib/workspace/application/doc/share_bloc.dart (hunk 1 above)
Comment (rileyhawk1417) [reply frontend/app_flowy/lib/workspace/application/doc/share_bloc.dart:44]: Ok let me get to work on that.

Comment (rileyhawk1417) [reply frontend/app_flowy/lib/workspace/application/doc/share_bloc.dart]: Finished applying the changes.

**Comment by rileyhawk1417** [comment]
I opened a new PR #518 that will manage sending notifications to the user. 
For now it only works when the user clicks the export button.
The plan is to have it pop up when the file has been written and if not it will throw an error.

**Pull request closed without merging**